shaders/EditorHexGrid.shader

```diff
@@ -1,27 +1,50 @@
 shader_type spatial;
 render_mode unshaded;
 
-// TODO: implement our own version of procedural hex grid tiling for a better
+// Implemented our own version of procedural hex grid tiling for a better
 // looking and more uniform hexes.
 // See andrewhungblog.wordpress.com/2018/07/28/shader-art-tutorial-hexagonal-grids/
 
-uniform sampler2D gridTexture;
 uniform sampler2D maskTexture;
 uniform vec4 color : hint_color;
 
 uniform vec2 gridSize;
 
-varying vec3 worldPos;
+const vec2 hexSize = vec2(1.7320508, 1); // 1.7320508 = sqrt(3)
+const float edgeLength = 1.3;
+
+varying vec2 worldPos;
+
+// Returns normalized distance to the nearest hex center.
+float calcHexCenterDistance(vec2 coord)
+{
+    vec4 hexCenter = round(vec4(coord, coord - vec2(1., .5)) / hexSize.xyxy);
+    vec4 offset = vec4(coord - hexCenter.xy * hexSize, coord - (hexCenter.zw + .5) * hexSize);
+    vec2 position = dot(offset.xy, offset.xy) < dot(offset.zw, offset.zw) ? abs(offset.xy) : abs(offset.zw);
+    return max(dot(position, hexSize * .5), position.y) + .5;
+}
 
 void vertex(){
-    worldPos = (WORLD_MATRIX * vec4(VERTEX, 1.0)).xyz;
+    worldPos = (CAMERA_MATRIX * vec4(0., 0., 0., 1.)).xz;
 }
 
 void fragment(){
-    vec4 gridTex = texture(gridTexture, (worldPos.xz / gridSize) *
-        vec2(2.778, 4.811));
+    // Antialiasing for small grids.
+    float aa = fwidth(VERTEX.xy).x;
```

Comment:
`fwidth` doesn't work on GLES2. To preserve as much compatibility with older devices, maybe an alternative calculation could be used here or another way would be for the shader to support the two approach: procedural and texture based, these two would be toggleable. The procedural grid would be the default and if GLES2 is detected then the shader switches to the texture grid, I think this can be done by setting a uniform bool variable through the microbe editor script.

Reply:
fwidth is only used to make the lines thicker when the camera is far away. So I can take the camera height for that.
The problem with GLES2 and the bitmap is, that it is non 2 square and repeating.

Reply:
Well the warning is about some implementations not working in GLES2 with those kind of repeating textures. It seems to not be a problem on desktop platforms, but some other platforms probably have issues and warranted Godot adding the warning.

And yeah, 100% the shader needs to work with GLES2 mode, or a workaround needs to be added.

Reply:
updated shader to work with GLES2